bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -116,6 +116,35 @@ spec:
 
           bracketify() { case "$1" in *:*) echo "[$1]" ;; *) echo "$1" ;; esac }
 
+          check_db_health() {
+            db="${1}"
+            serverid_file="${2}"
+
+            if [[ ! -f "${db}" ]]; then
+                  echo "${db} does not exist, skipping health check"
+                  return
+            fi
+            echo "Checking ${db} health"
+            serverid=$(ovsdb-tool show-log "${db}" | grep -m 1 -oP '(?<=server_id: )[[:xdigit:]]+')
+            # prev_servers is a list.  We are searching for " $server_id(" in the list
+            # prev_servers: 33b4("ssl:10.19.138.37:9643"), 720d("ssl:10.19.138.33:9643"), 7e4f("ssl:10.19.138.32:9643")
+            match=$(ovsdb-tool show-log "${db}" | grep "prev_servers:.* ${serverid}(" || true)
+            if [[ -z ${match} ]]; then
+                  echo "Current server_id $serverid not found in ${db} prev_servers list, checking servers"
+                  match=$(ovsdb-tool show-log "${db}" | grep "^ servers:.* ${serverid}(" || true)
+            fi
+            if [[ -z ${match} ]]; then
```

Comment:
I'm not sure this is a comprehensive way to check if the db is healthy. There could be previous log records of the db joining the cluster, but then it could become corrupt later. We need to check if ovsdb-server is actually up, and if we can make queries to this particular db. For example you could have:

[root@ovn-control-plane3 ~]# ovsdb-tool show-log /etc/openvswitch/ovnnb_db.db | grep servers
 servers: 23b1("tcp:[172.18.0.6]:6643"), 5c5d("tcp:[172.18.0.5]:6643"), f5e7("tcp:[172.18.0.3]:6643")
 servers: 23b1("tcp:[172.18.0.6]:6643"), 5c5d("tcp:[172.18.0.5]:6643")

Where now f5e7 has left the cluster, but your grep would still see it there and think everything is OK.

Reply:
That part was added as the result of the discussion in https://bugzilla.redhat.com/show_bug.cgi?id=1837953#c40 .
Basically, if the snapshot does not happen, the current id is not dumped in the prev_servers list and we attempt to look for it.
Also, not sure if what you are describing could happen (or is a remote corner case) as we are looking for the current serverid, which I don't expect to disappear from the server list.